src/workerd/io/worker-interface.capnp

```diff
@@ -136,6 +136,14 @@ struct Trace @0x8e8d911203762d34 {
     message @2 :Data;
   }
 
+  enum ExecutionModel {
+    stateless @0;
+    durableObject @1;
+    workflow @2;
+  }
+  executionModel @25 :ExecutionModel;
+  # the execution model of the worker being traced. Can be stateless for a regular worker,
+  # durableObject for a DO worker or workflow for the upcoming Workflows.
```

Comment:
Nit: can this be moved after `truncated` since it is using a higher ordinal value? It's already difficult enough to make sure we're not duplicating ordinals, having them out of order doesn't help ;-)